proteinflow/data/torch.py

```diff
@@ -701,6 +730,26 @@ def _process(self, filename, rewrite=False, max_length=None, min_cdr_length=None
                 data_entry.get_chain_id_array(chain_set)
             )
             out["chain_dict"] = data_entry.get_chain_id_dict(chain_set)
+            if self.load_ligands and len(ligands) != 0:
+                X_ligands = []
+                ligand_smiles = []
+                ligand_chains = []
+                for chain_i, chain in enumerate(chain_set):
```

Comment:
it would be nice to move this to `ProteinEntry` -- it has methods to get all the other relevant arrays concatenated for specified chains in alphabetic order